### ISSUE TITLE
Light: snmp test fails inside syslog-ng-devshell docker image

### ISSUE THREAD

**Issue by alltilla** [syslog-ng/syslog-ng#3183]
# syslog-ng
## Version of syslog-ng
Compiled from source (master: 531d70bfbcf19a97a5278f324d141386d568ce7f)

## Platform
 * `docker` image: `balabit/syslog-ng-devshell:latest`
 * `docker` runs on `ubuntu-xenial` or `ubuntu-bionic`

# Issue
## Failure
`test_snmp_dest_v2_cisco_ios_trap` fails during `make pytest-check`.
```
_________________________________________________________________________________________ test_snmp_dest_v2_cisco_ios_trap __________________________________________________________________________________________

config = <src.syslog_ng_config.syslog_ng_config.SyslogNgConfig object at 0x7f4e0ab9cd50>, syslog_ng = <src.syslog_ng.syslog_ng.SyslogNg object at 0x7f4e0ab9ca50>
snmptrapd = <src.helpers.snmptrapd.conftest.SNMPtrapd object at 0x7f4e0ab9c910>

    @pytest.mark.snmp
    def test_snmp_dest_v2_cisco_ios_trap(config, syslog_ng, snmptrapd):
        # create source driver config
        file_source = config.create_file_source(file_name="input.log")
    
        # create destination driver config
        snmp_objs = (
            "'{}', 'Octetstring', '{}'".format(OID1, CISCO_FACILITY),
            "'{}', 'Integer', '{}'".format(OID2, CISCO_SEVERITY),
            "'{}', 'Octetstring', '{}'".format(OID3, CISCO_MESSAGE_TYPE),
            "'{}', 'Octetstring', '{}'".format(OID4, CISCO_MESSAGE),
            "'{}', 'Timeticks', '{}'".format(OID5, CISCO_TIMETICKS),
        )
    
        trap_obj = "'.{}','Objectid','{}'".format(OID6, TRAP_DATA)
    
        snmp_destination = config.create_snmp_destination(
            host="'127.0.0.1'",
            port=snmptrapd.get_port(),
            version="'v2c'",
            snmp_obj=snmp_objs,
            trap_obj=trap_obj,
        )
    
        config.create_logpath(statements=[file_source, snmp_destination])
        config.update_global_options(keep_hostname="yes")
        file_source.write_log(input_log)
    
        syslog_ng.start(config)
    
>       snmptrap_logs = snmptrapd.get_logs()[2].rstrip().split("\t")
E       IndexError: list index out of range

config     = <src.syslog_ng_config.syslog_ng_config.SyslogNgConfig object at 0x7f4e0ab9cd50>
file_source = <src.syslog_ng_config.statements.sources.file_source.FileSource object at 0x7f4e0ab45410>
snmp_destination = <src.syslog_ng_config.statements.destinations.snmp_destination.SnmpDestination object at 0x7f4e0ab45310>
snmp_objs  = ("'1.3.6.1.4.1.9.9.41.1.2.3.1.2.55', 'Octetstring', 'SYS'",
 "'1.3.6.1.4.1.9.9.41.1.2.3.1.3.55', 'Integer', '6'",
 "'1.3.6.1.4.1.9.9.41.1.2.3.1.4.55', 'Octetstring', 'CONFIG_I'",
 "'1.3.6.1.4.1.9.9.41.1.2.3.1.5.55', 'Octetstring', 'Configured from console by vty1 (10.30.0.32)'",
 "'1.3.6.1.4.1.9.9.41.1.2.3.1.6.55', 'Timeticks', '97881'")
snmptrapd  = <src.helpers.snmptrapd.conftest.SNMPtrapd object at 0x7f4e0ab9c910>
syslog_ng  = <src.syslog_ng.syslog_ng.SyslogNg object at 0x7f4e0ab9ca50>
trap_obj   = "'.1.3.6.1.6.3.1.1.4.1.0','Objectid','.1.3.6.1.4.1.9.9.41.2.0.1'"

tests/python_functional/functional_tests/destination_drivers/snmp_destination/test_snmp_destination_v2c_cisco_ios_traps.py:83: IndexError
```

## Steps to reproduce
```
$ docker pull balabit/syslog-ng-devshell:latest
$ docker run -it balabit/syslog-ng-devshell:latest bash

# apt-get update && apt-get install -y snmptrapd
# git clone https://github.com/syslog-ng/syslog-ng.git
# cd syslog-ng/
# ./autogen.sh
# ./configure --prefix=$HOME/install/syslog-ng
# make
# make install
# make pytest-check
```

This is most probably a race-condition. Putting `time.sleep(1)` in different parts of the snmp test fixes the problem. Also, the `snmp-dest` works correctly, when tested manually.

## Input and output logs
[test_snmp_dest_v2_cisco_ios_trap.log](https://github.com/syslog-ng/syslog-ng/files/4358856/test_snmp_dest_v2_cisco_ios_trap.log)

**Comment by mitzkia** [comment]
@alltilla Thanks for the report and the reproduction steps. 
This should fix: #3185

**Comment by furiel** [comment]
Closed with #3185
